source/merge.ts

```diff
@@ -3,7 +3,7 @@ import is from '@sindresorhus/is';
 import {Options, Method, NextFunction, Instance, InterfaceWithDefaults} from './utils/types';
 import knownHookEvents, {Hooks, HookType, HookEvent} from './known-hook-events';
 
-export default function merge(target: Options, ...sources: Options[]): Options {
+export default function merge<Target extends { [key: string]: unknown }, Source extends { [key: string]: unknown }>(target: Target, ...sources: Source[]): Target & Source {
```

Comment:
I changed the type of this to better reflect that the merge can take an object of any arbitrary values.